jwst/assign_wcs/util.py

```diff
@@ -366,15 +366,32 @@ def compute_fiducial(wcslist, bounding_box=None):
     spatial_footprint = footprints[spatial_axes]
     spectral_footprint = footprints[spectral_axes]
 
+    n_input = len(wcslist)
     fiducial = np.empty(len(axes_types))
+    lon = np.empty(n_input * 2)
+    lat = np.empty(n_input * 2)
     if spatial_footprint.any():
-        lon, lat = spatial_footprint
+        for i in range(n_input):
+            wcs_init = wcslist[i]
+            domain_bounds = np.hstack([wcs_init.backward_transform(
+                *w.footprint(bounding_box=bounding_box).T) for w in wcslist])
+
+            output_bounding_box = []
+            for axis in wcslist[i].output_frame.axes_order:
+                axis_min, axis_max = domain_bounds[axis].min(), domain_bounds[axis].max()
+                output_bounding_box.append((axis_min, axis_max))
+
+            sky_bounds = np.array(wcs_init.forward_transform(*output_bounding_box))
+            lon[i * 2:(i + 1) * 2] = sky_bounds[0]
+            lat[i * 2:(i + 1) * 2] = sky_bounds[1]
```

Comment:
I think lines 374-386 recompute `spatial_footprint` in a roundabout way.
I would revert this code.